### PR TITLE
issue#7731: add endpoint for GCP objectstore client

### PR DESCRIPTION
**Pull request by chrislinan**

### CODE REVIEW

**Review by blackpiglet — APPROVED** [review]

**Comment by blackpiglet** [comment]
@chrislinan 
The go module configuration is not update to date.

**Comment by blackpiglet** [comment]
Thanks for the update. 
The code change looks good, but the Velero v1.14.0 waits for its RC.
Could we put this PR on hold until the GCP plugin's corresponding release is cut?

**Comment by chrislinan** [comment]
> Thanks for the update. The code change looks good, but the Velero v1.14.0 waits for its RC. Could we put this PR on hold until the GCP plugin's corresponding release is cut?

Yes, we could hold this PR. And I will open another PR for Velero to update documents and add this feature in docs.

File: velero-plugin-for-gcp/object_store.go
Comment (anshulahuja98) [inline velero-plugin-for-gcp/object_store.go:42]: should we maybe explicitly call it storeEndpoint?
That way if there are other endpoints which are required in future, we don't cause confusion.

**Review by anshulahuja98 — CHANGES_REQUESTED** [review]

File: velero-plugin-for-gcp/object_store.go
Comment (blackpiglet) [reply velero-plugin-for-gcp/object_store.go:42]: Good point. It's possible the snapshot endpoint also needs modification.

**Review by anshulahuja98 — APPROVED** [review]

**Comment by anshulahuja98** [comment]
@blackpiglet can you trigger the CI
I don't have permission on this repo

**Comment by chrislinan** [comment]
Hi @blackpiglet , Could you help review this PR? Thanks a lot.

**Comment by anshulahuja98** [comment]
@chrislinan - can you also make any relevant docs changes?

**Comment by anshulahuja98** [comment]
I don't have approver access on this repo. the PR LGTM otherwise.

**Comment by chrislinan** [comment]
> @chrislinan - can you also make any relevant docs changes?

Yes, I did some changes in backupstoragelocation.md

**Comment by chrislinan** [comment]
Hi @ywk253100 @blackpiglet @shubham-pampattiwar Could you help review this PR? Thanks.

**Comment by kaovilai** [comment]
I'm assuming they will review after v1.15 GA which is soon.

**Comment by blackpiglet** [comment]
> I'm assuming they will review after v1.15 GA which is soon.

Yeah, thanks, Tiger.

@chrislinan Sorry for the late response. I was working closely on the Velero v1.15.0 RC and GA.
Because we didn't consider testing this change in the v1.15.0 function test, it's better to merge this PR after new branch cut.

**Review by blackpiglet — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Pull request merged**
